### DISCUSSION TITLE
3D Gaming Engine

### DISCUSSION THREAD

**Discussion by gladiatormedia** [dotnet/maui#7973] [General]
Hi All.  I'm interested in creating an app the utilizes a 3D game engine on .NET MAUI.  This is a project that I have never done before, so I relying on the support of the community to steer me in the right direction.  I am trying to build something like this https://www.youtube.com/watch?v=cnI_Au52Ocs.  Some of the key elements that I need are as follows... 

1. Allowing multiple players to work within the same design. You can see this in the link above.  Essentially, a project is created in the cloud, and multiple team members can be working within the same space.
2. The use of Constructive Solid Geometry, I will need this to construct wood parts with holes and grooves in them.

I was thinking about using Babylonjs inside a Maui Blazor app.  Is that possible?  

If it is possible, what types of issues do you foresee running into? if any.

My target platforms are Windows Desktop, iOS, and android.

Is there a better solution?

**Comment by DavidBerschauer** [comment]
BabylonJS inside Blazor Hybrid is working: [EventHorizon.Maui.BabylonJS](https://github.com/canhorn/EventHorizon.Blazor.TypeScript.Interop.Generator/tree/main/Sample/EventHorizon.Maui.BabylonJS)
